### PR TITLE
[Bugfix] Restore Missing ctests

### PR DESCRIPTION
**Pull request by cderb**
Fix for issue #2636 
Re-adding cpp files while waiting changes for gtest.

### CODE REVIEW

**Comment by junliume** [comment]
@cderb there is a HIP tidy issue, has it been fixed?
```
[2024-01-05T07:41:40.725Z] In file included from /var/jenkins/workspace/MLLIBS_MIOpen_PR-2649/test/immed_conv3d.cpp:26:

[2024-01-05T07:41:40.725Z] /var/jenkins/workspace/MLLIBS_MIOpen_PR-2649/test/immed_conv3d.hpp:62:28: error: use of undeclared identifier 'testing'

[2024-01-05T07:41:40.725Z] class Conv3dFloat : public testing::TestWithParam<std::vector<std::string>>

[2024-01-05T07:41:40.725Z]                            ^
```

**Comment by cderb** [comment]
Addressed the tidy issue just now.

File: test/pooling2d.hpp [hunk 1]
```diff
@@ -2,7 +2,7 @@
  *
  * MIT License
  *
- * Copyright (c) 2019 Advanced Micro Devices, Inc.
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
```
Comment (JehandadKhan) [inline test/pooling2d.hpp]: I dont think this is required, since you are not touching the rest of the file.

Comment (cderb) [reply test/pooling2d.hpp]: There are a few minor lines added to the end of this file in a previous PR for gtest support. I'm not certain it's enough to change the date here.

File: test/gtest/immed_conv3d_codecov.cpp [hunk 2]
```diff
@@ -28,45 +28,27 @@
 #include "get_handle.hpp"
 #include "test_env.hpp"
 
-#include "conv3d.hpp"
+#include "immed_conv3d.hpp"
```
Comment (atamazov) [inline test/gtest/immed_conv3d_codecov.cpp:31]: [Q] A bugfix?

**Review by atamazov — CHANGES_REQUESTED** [review]
A couple of questions + a pair of leftovers:

- [x] `test` should not depend on `test/gtest`. Please remove `#include <gtest/gtest.h>` from `test/conv3d.hpp` and `test/pooling2d.hpp`. (Originated from https://github.com/ROCm/MIOpen/pull/2599#discussion_r1446639588 and https://github.com/ROCm/MIOpen/pull/2599#discussion_r1446640391)

- [x] Please also resolve https://github.com/ROCm/MIOpen/pull/2599#pullrequestreview-1812070818 right in this PR. Thanks!

File: test/gtest/immed_conv3d_codecov.cpp (hunk 2 above)
Comment (cderb) [reply test/gtest/immed_conv3d_codecov.cpp:31]: yes, previous header had a different definition of conv3d_driver.

File: test/pooling2d.hpp (hunk 1 above)
Comment (cderb) [reply test/pooling2d.hpp]: reverted

**Review by atamazov — APPROVED** [review]
LGTM!

**Review by junliume — APPROVED** [review]

**Pull request merged**